enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,192 @@
+* With rpm-ostree, a problem we encountered is that the immutable CoreOS model guards the immutability of certain directories. With that being said, if we write a file (e.g. SSH key) to an immutable directory during an OS image build and use rpm-ostree to rebase onto the newly built OS image, that file will not actually be there when the node reboots. Will this be handled/considered in the future planning of RHCOS and bootc? 
+* Questions raised alongside “moving to layered update path by default”:
+    - Should all config be built into the image? 
+    - What will the update path look like if certain config changes are not baked into the image because (1) We normally don’t write them through MachineConfig (e.g. cert writings) (2) Less reboot is favored by Node Disruption Policies?
+    - Certain config changes that the MCO watches don't trigger reboots, but it remains unknown if moving to layered update path by default can promise this too.
+
+## Test Plan
+
+* Phase 1: Please refer to On Cluster Layering enhancement for more details  
+* Phase 2: TBD 
+* Phase 3 : Bootc Switch Over 
+    - Develop an e2e test suite and run payloads with bootc: Tests should target the TestOSImageURL, TestKernelArguments, TestKernelType and TestExtensions. Results will be used to discuss whether bootc is ready to perform rpm-ostree’s tasks.
+    - Create a bootc experimentation sandbox by having the BootcNodeManagement FG: When the FG is on, the rpm-ostree commands will be replaced by equivalent bootc commands to manually test the capability of bootc in non-layered machine configuration workflow. Switch back to rpm-ostree as the default option when not compatible. This experimentation prepares for Scenario 1 in the following section. 
+    - Test bootc with On Cluster Layering: When the BootcNodeManagement and OnClusterBuild FG is on, On Cluster Layering will be the default path for machine configuration, accompanied by bootc who is responsible for image roll out. This test prepares for Scenario 2 in the following section.  
+
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+There are three cases we should consider and try to support under tech preview
+  * Scenario 1: We don’t have layered update path as the default, but we start to provide rpm-ostree + bootc dual support (likely in 4.17 and 4.18)
+  * Scenario 2: We now have layered update path as the default and we still have rpm-ostree + bootc dual support (likely in 4.19)
+  * Scenario 3: We have completed the switch over to layered update path + bootc for image management + dnf for package management (likely in  4.20)
+
+### Tech Preview -> GA
+
+* Unit and E2E test coverage for the two criteria mentioned above 
+* Unit and E2E test coverage for the two criteria mentioned above 
+* The GA of this enhancement is also dependent on:
+    - On Cluster Layering GA 
+    - Move to Layered Update Path by Default GA
+    - Split RHCOS into Layers & Image Compose 
```

Comment:
Should be orthogonal

Reply:
Still outstanding. I don't think this needs to be a hard requirement for GA.

Reply:
Fixed